### DISCUSSION TITLE
Attempting to upgrade from mdx 1.6.22 to 2.1.2 throwing error: cannot find module estree-walker.

### DISCUSSION THREAD

**Discussion by ajaypillay** [mdx-js/mdx#2088] [Q&A] [answered]
Hi, I've been attempting to ugprade this Meteor package [Meteor-mdx](https://github.com/Apollinaire/meteor-mdx) from MDX 1.6.22 to MDX 2.1.2.

My current `package.js` file is as such:
```js
var packages = ["ecmascript", "isobuild:compiler-plugin@1.0.0"]

Package.describe({
	name: "foo",
	version: "0.0.5",
	summary: "Compiler plugin that supports MDX files in Meteor",
	git: "https://github.com/Apollinaire/meteor-mdx",
	documentation: "README.md"
})

Package.registerBuildPlugin({
	name: "compile-mdx",
	use: ["ecmascript", "babel-runtime", "babel-compiler", "caching-compiler"],
	sources: ["compiler.js", "plugin.js"],
	npmDependencies: {
          /* "estree-walker": "3.0.0", */ // Tried adding this in but it didn't change anything
          "@babel/runtime": "7.18.9",
          "@mdx-js/mdx": "2.1.2"
	}
})

Package.onUse(function use(api) {
	api.versionsFrom("1.3.2.4")
	api.use(packages, ["server", "client"])
})
```

And in my `compiler.js` file, including any line like `import mdx from "@mdx-js/mdx"` results in:
```
=> Errors prevented startup:                  
   
   While loading plugin `compile-mdx` from package `foo`:
   internal/modules/cjs/loader.js:902:15: Cannot find module '/home/.../myProject/packages/foo/.npm/plugin/compile-mdx/node_modules/estree-walker'
```
I tried manually including `estree-walker` as shown above but it resulted in the same error.

How should I go about troubleshooting this?

**Comment by ChristianMurphy** [comment, marked as answer]
`meteor-mdx` doesn't support MDX 2 yet.
It is calling APIs which don't exist in version 2 https://github.com/Apollinaire/meteor-mdx/blob/0cee4c6ecc664c8a0341fee0cb1dff324b4fa14f/compiler.js#L70 (version 2 API https://mdxjs.com/packages/mdx/)

The dependency issue is just one side effect of it not being designed for version 2.
This would be something to take up with the author(s) of the integration
https://github.com/Apollinaire/meteor-mdx/issues

**Reply by ChristianMurphy** [reply]
There may also be a deeper issue with meteor not supporting ESM yet (see: https://github.com/meteor/meteor/discussions/11727, https://github.com/meteor/meteor/issues/11631, and https://github.com/meteor/meteor/issues/10906)
Which is something the Meteor team may be able to answer questions around and resolve.

**Reply by ajaypillay** [reply]
Oh yes, I'm aware of the need to update `mdx.sync` to `await compile()`, which is part of what I was intending to upgrade. I'll need to look into the ESM bit, thanks.
